lib/galaxy/tools/actions/upload_common.py

```diff
@@ -282,8 +281,10 @@ def create_paramfile( trans, uploaded_datasets ):
     """
     def _chown( path ):
         try:
-            pwent = pwd.getpwnam( trans.user.email.split('@')[0] )
-            cmd = [ '/usr/bin/sudo', '-E', trans.app.config.external_chown_script, path, pwent[0], str( pwent[3] ) ]
+            # get username from email/username
+            pwent = trans.user.system_user_pwent(trans.app.config.real_system_username)
+            cmd = trans.app.config.external_chown_script.split()
```

Comment:
same here, `shlex.split(trans.app.config.external_chown_script)` would be preferable.